packages/common/src/directives/ng_switch.ts

```diff
@@ -189,24 +190,32 @@ export class NgSwitch {
  */
 @Directive({selector: '[ngSwitchCase]'})
 export class NgSwitchCase implements DoCheck {
-  private _view: SwitchView;
+  private _view?: SwitchView;
   /**
    * Stores the HTML template to be selected on match.
    */
   @Input() ngSwitchCase: any;
 
   constructor(
       viewContainer: ViewContainerRef, templateRef: TemplateRef<Object>,
-      @Host() private ngSwitch: NgSwitch) {
-    ngSwitch._addCase();
-    this._view = new SwitchView(viewContainer, templateRef);
+      @Optional() @Host() private ngSwitch: NgSwitch) {
+    if (!ngSwitch) {
+      if ((typeof ngDevMode === 'undefined' || ngDevMode) && isDevMode()) {
+        throwNgSwitchProviderNotFoundError('NgSwitchCase')
+      }
+    } else {
+      ngSwitch._addCase();
+      this._view = new SwitchView(viewContainer, templateRef);
+    }
```

Comment:
I don't think we need to test for `ngSwitch` in production mode. It will just fail when we try to call `_addCase()`.

Also, I don't believe we need `_view` to be optional, since if `ngSwitch` is not valid this whole class will never complete instantiation.

```suggestion
if ((typeof ngDevMode === 'undefined' || ngDevMode) && isDevMode()) {
  if (!ngSwitch) {
    throwNgSwitchProviderNotFoundError('NgSwitchCase');
  }
}
ngSwitch._addCase();
this._view = new SwitchView(viewContainer, templateRef);
```

Reply:
Thanks, I have change code according to your recommendations.